src/apps/xcalibur/arbitrum/xcalibur.pool.token-fetcher.ts

```diff
@@ -0,0 +1,98 @@
+import { Inject } from '@nestjs/common';
+import { BigNumber, BigNumberish } from 'ethers';
+import { APP_TOOLKIT, IAppToolkit } from '~app-toolkit/app-toolkit.interface';
+import { PositionTemplate } from '~app-toolkit/decorators/position-template.decorator';
+import { Erc20 } from '~contract/contracts';
+import { AppTokenTemplatePositionFetcher } from '~position/template/app-token.template.position-fetcher';
+import {
+  GetAddressesParams,
+  DefaultAppTokenDefinition,
+  GetUnderlyingTokensParams,
+  UnderlyingTokenDefinition,
+  GetPricePerShareParams,
+  DefaultAppTokenDataProps,
+} from '~position/template/app-token.template.types';
+
+import { XcaliburContractFactory, XcaliburSwapFactory, XcaliburRouter, UniswapPairV2} from '../contracts';
+
+@PositionTemplate()
+export class ArbitrumXcaliburPoolTokenFetcher extends AppTokenTemplatePositionFetcher<XcaliburSwapFactory> {
```

Comment:
`AppTokenTemplatePositionFetcher` requires the provided contract to be an ERC20 as it will be looking for symbol, decimals and totalSupply